src/main/java/slimeknights/tconstruct/plugin/jei/ToolBuildingCategory.java

```diff
@@ -0,0 +1,125 @@
+  private static final int ITEM_SIZE = 16;
+
+  public ToolBuildingCategory(IGuiHelper guiHelper) {
+    this.icon = guiHelper.createDrawableItemStack(TinkerTools.pickaxe.get().getRenderTool());
+    this.background = guiHelper.createDrawable(BACKGROUND_LOC, 128, 77, WIDTH, HEIGHT);
+    this.slot = guiHelper.createDrawable(BACKGROUND_LOC, 24, 14, SLOT_SIZE, SLOT_SIZE);
+    this.anvil = guiHelper.createDrawable(BACKGROUND_LOC, 128, 61, ITEM_SIZE, ITEM_SIZE);
+  }
+
+  @Override
+  public void setRecipe(IRecipeLayoutBuilder builder, ToolBuildingRecipe recipe, IFocusGroup focuses) {
+    recipe.getIngredients().forEach(ingredient -> builder.addInvisibleIngredients(RecipeIngredientRole.INPUT).addIngredients(ingredient));
+    recipe.getExtraRequirements().forEach(ingredient -> builder.addSlot(RecipeIngredientRole.INPUT, 0, 0).addIngredients(ingredient));
+
+    List<LayoutSlot> slots = StationSlotLayoutLoader.getInstance().get(recipe.getSlotId()).getInputSlots();
+    List<ItemStack> items = recipe.getIngredients().stream().flatMap(ingredient -> Arrays.stream(ingredient.getItems())).toList();
+
+    if (items.size() != slots.size()) {
+      TConstruct.LOG.error("Part count and slot count for %s do not match!", recipe.getId().toString());
+    }
+
+    Vec2 offsets = getOffsets(slots);
+    for (int i = 0; i < items.size(); i++) {
+      builder.addSlot(RecipeIngredientRole.INPUT, (int) (slots.get(i).getX() + offsets.x), (int) (slots.get(i).getY() + offsets.y)).addItemStack(items.get(i));
+    }
+
+    builder.addSlot(RecipeIngredientRole.OUTPUT, WIDTH - 23, 23).addItemStack(recipe.getOutput().asItem().getDefaultInstance());
+  }
+
+  @Override
+  public void draw(ToolBuildingRecipe recipe, IRecipeSlotsView recipeSlotsView, PoseStack stack, double mouseX, double mouseY) {
+    if (isBroadTool(recipe)) {
+      this.anvil.draw(stack, 73, 42);
+    }
+    List<LayoutSlot> slots = StationSlotLayoutLoader.getInstance().get(recipe.getId()).getInputSlots();
+    Vec2 offsets = getOffsets(slots);
+    for (LayoutSlot slot : slots) {
+      // need to offset by 1 because the inventory slot icons are 18x18
+      this.slot.draw(stack, (int) (slot.getX() + offsets.x - 1), (int) (slot.getY() + offsets.y - 1));
+    }
+  }
+
+  @Override
+  public List<Component> getTooltipStrings(ToolBuildingRecipe recipe, IRecipeSlotsView recipeSlotsView, double mouseX, double mouseY) {
+    return isBroadTool(recipe) && GuiUtil.isHovered((int) mouseX, (int) mouseY, 73, 42, ITEM_SIZE, ITEM_SIZE) ?
+      List.of(TConstruct.makeTranslation("jei", "tinkering.tool_recipes.anvil")) :
+      List.of();
+  }
+
+  private boolean isBroadTool(ToolBuildingRecipe recipe) {
+    return recipe.getIngredients().size() >= 4;
+  }
+
+  private Vec2 getOffsets(List<LayoutSlot> slots) {
+    int minX, maxX, minY, maxY;
+    minX = slots.get(0).getX();
+    maxX = slots.get(0).getX();
+    minY = slots.get(0).getY();
+    maxY = slots.get(0).getY();
```

Comment:
I asked earlier, why are the offsets needed? The slot positions work fine in the anvil.

Reply:
My texture is smaller than the station/anvil texture so I can't just use the layout slot positions. I could resize my texture to be the same size as station/anvil texture if you would like.

Reply:
I think that would be preferable. Offsetting the slots dynamically like this suggests they are wrongly placed in the original UI; which perhaps they are, if its just a single outlier then just fix the outlier. I am not against a static offset (same numbers used for all layouts).

In the future, we may want to set up some validation bounds on them in the future now that we have multiple distinct UIs using the numbers. But that's out of scope of this PR for now.